webapp/packages/plugin-settings-panel/src/SettingsPanel/Setting.tsx

```diff
@@ -20,16 +19,27 @@ interface Props {
 
 export const Setting = observer<Props>(function Setting({ source, setting }) {
   const translate = useTranslate();
-  const pluginManagerService = useService(PluginManagerService);
-
-  const defaultValue = pluginManagerService.getSettings(setting.scope)?.getDefaultValue(setting.key);
+  const settingsProviderService = useService(SettingsProviderService);
+  const settingProvider = settingsProviderService.getSettings(setting.scope);
 
   const key = `${setting.scope}.${setting.key}`;
-  const value = source.getValue(key) ?? defaultValue ?? '';
+  let value = source.getEditedValue(key);
```

Comment:
we can incapsulate the getting `value` logic into function `getValue` so our render method looks a bit cleaner